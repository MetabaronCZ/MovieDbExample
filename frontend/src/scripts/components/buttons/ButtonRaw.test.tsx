import { act } from '@testing-library/react';
import { describe, expect, it, jest } from '@jest/globals';
import { fireEvent, getByRole, getByText } from '@testing-library/dom';

import { ButtonRaw } from './ButtonRaw';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/button/ButtonRaw', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(
      <ButtonRaw>CONTENT</ButtonRaw>,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const button = getByText(container, 'CONTENT');
    expect(button).toBeInTheDocument();
    expect(button.tagName.toLowerCase()).toEqual('button');
  });

  it('should render props', async () => {
    const { container } = await waitForComponent(
      <ButtonRaw
        id="ID"
        className="CLASSNAME"
        title="TITLE"
        type="submit"
        disabled
      >
        CONTENT
      </ButtonRaw>,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const button = getByText(container, 'CONTENT');
    expect(button).toBeInTheDocument();

    expect(button).toHaveAttribute('id', 'ID');
    expect(button).toHaveClass('CLASSNAME');
    expect(button).toHaveAttribute('title', 'TITLE');
    expect(button).toHaveAttribute('type', 'submit');
    expect(button).toHaveAttribute('disabled', '');
  });

  it('should be clickable', async () => {
    const onClick = jest.fn();

    const { container } = await waitForComponent(
      <ButtonRaw onClick={onClick}>CONTENT</ButtonRaw>,
      { wrapper: TestComponentWrapper },
    );
    expect(onClick).not.toHaveBeenCalled();

    const button = getByRole(container, 'button');
    expect(button).toBeInTheDocument();

    // click on button
    act(() => {
      fireEvent.click(button);
    });
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('should not be clickable when disabled', async () => {
    const onClick = jest.fn();

    const { container } = await waitForComponent(
      <ButtonRaw onClick={onClick} disabled>
        CONTENT
      </ButtonRaw>,
      { wrapper: TestComponentWrapper },
    );
    expect(onClick).not.toHaveBeenCalled();

    const button = getByRole(container, 'button');
    expect(button).toBeInTheDocument();

    // click on button
    act(() => {
      fireEvent.click(button);
    });
    expect(onClick).not.toHaveBeenCalled();
  });
});
