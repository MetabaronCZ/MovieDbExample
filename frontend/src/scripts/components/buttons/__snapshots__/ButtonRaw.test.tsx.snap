// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/button/ButtonRaw should render properly 1`] = `
.c0 {
  display: block;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  cursor: not-allowed;
}

<div>
  <button
    class="c0"
    type="button"
  >
    CONTENT
  </button>
</div>
`;

exports[`components/button/ButtonRaw should render props 1`] = `
.c0 {
  display: block;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  cursor: not-allowed;
}

<div>
  <button
    class="c0 CLASSNAME"
    disabled=""
    id="ID"
    title="TITLE"
    type="submit"
  >
    CONTENT
  </button>
</div>
`;
